File: election-api-scala/app/service/ResultService.scala
package service

import com.google.inject.ImplementedBy
import model.ConstituencyResult

import scala.util.Try

@ImplementedBy(classOf[MapResultService])
trait ResultService {
  def GetResult(id: Int): Option[ConstituencyResult]
  def NewResult(result: ConstituencyResult): Try[Unit]
  def GetAll: Seq[ConstituencyResult]
  def reset(): Try[Unit]
}

File: election-api-java/src/main/java/bbc/news/elections/service/ResultService.java
package bbc.news.elections.service;

import bbc.news.elections.model.ConstituencyResult;

import java.util.Map;

public interface ResultService {
    ConstituencyResult getResult(Integer id);
    void newResult(ConstituencyResult result);
    Map<Integer,ConstituencyResult> getAll();
    void reset();
}
